lib/iris/fileformats/netcdf.py

```diff
@@ -2865,6 +2865,13 @@ def _increment_name(self, varname):
 
     @staticmethod
     def _lazy_stream_data(data, fill_value, fill_warn, cf_var):
+        if hasattr(data, "shape") and data.shape == (1,) + cf_var.shape:
+            # Reduce dimensionality where bounds data is for a scalar point -
+            #  bounds data is 2D but contains just 1 row, which causes
+            #  broadcast ambiguity between the data and its 1D cf_var.
+            # (Don't do this check for string data).
+            data = np.atleast_1d(data[0])
```

Comment:
@wjbenfold tells me that numpy `squeeze` can do this.

Reply:
Well it can, but it would reduce "any" length-1 dims.
So I'd say the existing code is clearer in terms of exactly what we need to do here.

But if we're now avoiding that 'string list' case, do we still need the `atleast_1d` call ?
It would be simpler still if we avoid the need for that.

Reply:
d786c7ac2c468b7b429a0066358c9ec67fae760d

Reply:
`numpy.squeeze` takes the `axis` keyword.  Would `np.squeeze(data, axis=0)` do what you need?

Reply:
Thanks all for your help this morning.

### Cases where `data[0]` creates a single, non-array value

@pp-mo wasn't expecting this, so this needed investigating to check nothing nasty is happening.

For scalar coordinates, when streaming the point data we get `data.shape == (1,)` and `cf_var.shape == ()`, which meets this check:

https://github.com/SciTools/iris/blob/19efa6582894bf4a9e3773463b40e6e8a0905c03/lib/iris/fileformats/netcdf.py#L2868

### Correct use of `squeeze()`

Considering the above, I still think the correct solution is to include insurance that the `data` will still be an array, even after removing a dimension. But I appreciate now that `data.squeeze()` could end up removing other axes that we weren't planning to mess with. So @rcomer's suggestion to include `axis=0` seems like the right fix - the `if` line makes sure we don't end up making some other, unexpected modification to `data`. 19efa6582894bf4a9e3773463b40e6e8a0905c03